api/src/main/java/io/minio/MinioClient.java

```diff
@@ -2647,49 +2715,45 @@ public void composeObject(
       }
     }
 
-    if (partsCount == 1) {
+    if (args.sources().size() == 1) {
```

Comment:
1. You would need to do `if (args.sources.size() == 1 && src.offset() == null && src.length() == null) {`
2. No need to use `sources.get(0)` because `src` is already pointing to `sources.get(0)`